department-of-veterans-affairs/vets-api
# API-34447-va-notify-poa-updated

## Summary
* Adds the job to send the notification email
    * Send the accepted emails for rep and org flow (2122a and 2122)
* Adds tests for email sending and templates as well as the helper methods
* Adds request ICN to auth_headers so we can access it in the notification email
* Calls job from poa_updater when that is successful
* This workflow only applies to v2 POA

## Related issue(s)
[API-40377](https://jira.devops.va.gov/browse/API-40377)

## Testing done


#### Testing Notes
switching the job to use new.perform is not strictly required, but wll make it easier to step through and see.
* you will need to switch out so the template sends to your email: `email_address: 'you.email@here'` replaces `recipient_identifier: icn_for_vanotify(poa.auth_headers),` for whichever you are sending
* Change line 91 in the `power_of_attorney/base_controller` to use `new.perform`
* Change line 32 in the v2 PoaFormBuilderJob to use `new.perform` 
* Change line 31 in poa_updater to use `new.perform`
* Submit a v2 2122 or 2122a.  Once it has run all the way through the job you should see an email pretty quick.  The titles are: ORG: `We processed your form to appoint a VSO` and Rep: `We processed your form to appoint an accredited representative`

## Screenshots


## What areas of the site does it impact?
* modified: config/settings/test.yml
* modified: modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/base_controller.rb
* modified: modules/claims_api/app/sidekiq/claims_api/poa_updater.rb
* modified: modules/claims_api/app/sidekiq/claims_api/v2/poa_form_builder_job.rb
* new: modules/claims_api/app/sidekiq/claims_api/va_notify_job.rb
* new: modules/claims_api/spec/sidekiq/va_notify_job_spec.rb

## Acceptance criteria

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
